src/AutoUIApp/ActionSidebar.tsx

```diff
@@ -29,16 +30,26 @@ const replaceRefValues = (
 
 	const build: any = {};
 	for (const key in object) {
-		if (object.hasOwnProperty(key)) {
-			let value = object[key as keyof JSONSchema] as JSONSchema;
+		if (Object.prototype.hasOwnProperty.call(object, key)) {
+			let value = object[key as keyof JSONSchema];
 
 			// If this is an object, containing the $ref, evaluate it
-			if (typeof value === 'object' && !!value['$ref']) {
+			if (
+				typeof value === 'object' &&
+				!Array.isArray(value) &&
+				isJSONSchema(value) &&
+				!!value['$ref']
+			) {
 				value = getFromRef(openApiJson, value['$ref']);
 			}
 
 			// If this is an object without $ref, recurse
-			if (typeof value === 'object' && !value['$ref']) {
+			if (
+				typeof value === 'object' &&
+				!Array.isArray(value) &&
+				isJSONSchema(value) &&
+				!value['$ref']
+			) {
```

Comment:
Just making sure: intentionally not if else?

Reply:
this is not an if else, as there are other possible cases: value is not an object, value is not an array, value is not a JSONSchema...